s3api/controllers/base.go

```diff
@@ -92,7 +92,7 @@ func (c S3ApiController) GetActions(ctx *fiber.Ctx) error {
 	}
 
 	if uploadId != "" {
-		if maxParts < 0 || (maxParts == 0 && ctx.Query("max-parts") != "") {
+		if (maxParts < 0 && ctx.Request().URI().QueryArgs().Has("max-parts")) || maxParts > 2147483647 {
```

Comment:
I removed this check: it's unnecessary. If max-parts is larger than math.MaxInt32 it will result the variable value to be -1 and it will return the error